### PR TITLE
Replace search form smarty template by twig component

### LINKED ISSUE #33085: Twig Component : Search form, only the header form
See #33101 for description

This is the part regarding search form

### PR DESCRIPTION
**Pull request by lartist**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Replace search form smarty template by twig component
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI 🟢 and auto tests 🟢 
| Fixed ticket?     | Fixes #33085 
| Related PRs       |
| Sponsor company   |

### CODE REVIEW

**Review by boherm — DISMISSED** [review]

**Review by M0rgan01 — DISMISSED** [review]

**Review by tleon — DISMISSED** [review]

**Review by hibatallahAouadni — COMMENTED** [review]
Hello @lartist 

There's some conflits :disappointed: could you please resolve it :pray: 

Thanks!

**Comment by lartist** [comment]
> Hello @lartist
> 
> There's some conflits 😞 could you please resolve it 🙏
> 
> Thanks!

Hi @hibatallahAouadni , yes sure. I'm waiting for this one first https://github.com/PrestaShop/PrestaShop/pull/33173 otherwise I would have to rebase twice

**Comment by lartist** [comment]
https://github.com/lartist/ga.tests.ui.pr/actions/runs/5543220517

**Review by boherm — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *#}
+<form id="header_search"
+      class="bo_search_form dropdown-form js-dropdown-form collapsed"
+      method="post"
+      action="{{ link.getAdminLink('AdminSearch') }}"
+      role="search">
+  <input type="hidden" name="bo_search_type" id="bo_search_type" class="js-search-type" />
+  {% if show_clear_btn is defined and show_clear_btn %}
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig]: Don't you need to inject `show_clear_btn` as well then?

File: src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        <a class="dropdown-item" data-item="{{ 'Modules'|trans({}, 'Admin.Global') }}" href="#" data-value="7" data-placeholder="{{ 'Module name'|trans({}, 'Admin.Navigation.Header') }}" data-icon="icon-puzzle-piece"><i class="material-icons">extension</i> {{ 'Modules'|trans({}, 'Admin.Global') }}</a>
+      </div>
+      <button class="btn btn-primary" type="submit"><span class="d-none">{{ 'SEARCH'|trans({}, 'Admin.Navigation.Header') }}</span><i class="material-icons">search</i></button>
+    </div>
+  </div>
+</form>
+
+<script type="text/javascript">
+  $(function(){
+    {% if search_type is defined and search_type %}
```
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig]: SAme here seems like `search_type` should be injected, even if it's not always present in smart global vars

File: src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig (hunk 1 above)
Comment (lartist) [reply src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig]: you're right

File: src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig (hunk 2 above)
Comment (lartist) [reply src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig]: nice catch

File: admin-dev/themes/new-theme/template/layout.tpl [hunk 3]
```diff
@@ -34,11 +34,14 @@
       <div class="component component-search" id="header-search-container">
         <div class="component-search-body">
           <div class="component-search-top">
+
```
Comment (boherm) [inline admin-dev/themes/new-theme/template/layout.tpl:37]: I'm not blocking for this 😉
```suggestion
```

**Review by boherm — APPROVED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  {% if showClearBtn is defined and showClearBtn %}
+    <a href="#" class="clear_search hide"><i class="icon-remove"></i></a>
+  {% endif %}
+  <div class="input-group">
+    <input type="text" class="form-control js-form-search" id="bo_query" name="bo_query" value="{{ boQuery }}" placeholder="{{ 'Search (e.g.: product reference, customer name…)'|trans({}, 'Admin.Navigation.Header') }}" aria-label="{{ 'Searchbar'|trans({}, 'Admin.Navigation.Header') }}">
+    <div class="input-group-append">
+      <button type="button" class="btn btn-outline-secondary dropdown-toggle js-dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+        {{ 'Everywhere'|trans({}, 'Admin.Navigation.Header') }}
+      </button>
+      <div class="dropdown-menu js-items-list">
```
Comment (matks) [inline src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig:40]: I would suggest some blank lines to make it easier to read :smile:

**Review by matks — APPROVED** [review]
One comment not blocking, else LGTM 😉

**Review by tleon — APPROVED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig (hunk 4 above)
Comment (lartist) [reply src/PrestaShopBundle/Resources/views/Admin/Component/Layout/search_form.html.twig:40]: Ok I'll keep it in mind for the next time. 👍

**Comment by lartist** [comment]
https://github.com/lartist/ga.tests.ui.pr/actions/runs/5589473537

**Comment by mflasquin** [comment]
thanks @lartist

**Pull request merged**